src/features/legend_panel/components/LegendPanel/LegendPanel.tsx

```diff
@@ -18,54 +18,38 @@ interface LegendPanelProps {
 }
 
 export function LegendPanel({ layers, iconsContainerRef }: LegendPanelProps) {
-  // isOpen - is an overall state of panel. Was closed stores wether the panel was closed by user -> preferred to be closed
   const [isOpen, setIsOpen] = useState<boolean>(false);
-  const [wasClosed, setWasClosed] = useState<boolean>(true);
 
-  const isMobile = useMediaQuery(IS_MOBILE_QUERY);
   const turnOffTooltip = useAction(currentTooltipAtom.turnOffById);
 
-  useEffect(() => {
-    if (isMobile) {
-      setIsOpen(false);
-      turnOffTooltip(LEGEND_PANEL_FEATURE_ID);
-    }
-  }, [isMobile]);
-
-  useEffect(() => {
-    if (layers.length && !isMobile && !wasClosed && !isOpen) {
-      setIsOpen(true);
-    } else if (!layers.length && isOpen) {
-      setIsOpen(false);
-    }
-  }, [layers]);
-
   const onPanelClose = useCallback(() => {
     setIsOpen(false);
-    setWasClosed(true);
+    turnOffTooltip(LEGEND_PANEL_FEATURE_ID);
   }, [setIsOpen]);
 
   const onPanelOpen = useCallback(() => {
     setIsOpen(true);
-    setWasClosed(false);
   }, [setIsOpen]);
 
   return (
     <>
-      <Panel
-        header={<Text type="heading-l">{i18n.t('legend')}</Text>}
-        onClose={onPanelClose}
-        className={clsx(s.sidePanel, isOpen && s.show, !isOpen && s.hide)}
-        classes={{
-          header: s.header,
-        }}
-      >
-        <div className={s.panelBody}>
-          {layers.map((layer) => (
-            <LegendsList layer={layer} key={layer.id} />
-          ))}
-        </div>
-      </Panel>
+      <PanelWrap onPanelClose={onPanelClose} isPanelOpen={isOpen}>
+        <Panel
+          header={<Text type="heading-l">{i18n.t('legend')}</Text>}
+          onClose={onPanelClose}
+          className={clsx(s.legendPanel, isOpen && s.show, !isOpen && s.hide)}
+          classes={{
```

Comment:
as far as I remember if you pass {} to prop it will be new object on every render

Reply:
I mean you can do 

const legendPanelClasses = { header: s.header}
somewhere on the top of file, and link to object will be the same all the time

Reply:
we should do it next time then :) this PR takes too long already